src/reference/config/bindings/asyncapi/client.md

```diff
@@ -0,0 +1,173 @@
+---
+shortTitle: client
+---
+
+# asyncapi client
+
+The asyncapi client binding creates composite of `kafka` or `mqtt` or `http`, and `tls`, `tcp` bindings with client kind and adapts AsyncAPI streams to Kafka/MQTT/HTTP streams.
+
+```yaml {3}
+<!-- @include: ./.partials/client.yaml -->
+```
+
+## Configuration (\* required)
+
+<!-- @include: ../.partials/vault.md -->
+
+### options
+
+> `object`
+
+`client`-specific options.
+
+```yaml
+specs:
+  http_api:
+    servers:
+      - name: plain
+    catalog:
+      my_catalog:
+        subject: petstore
+        version: latest
+```
+
+<!-- @include: ./.partials/options.md -->
+
+### options.kafka
+
+> `object`
+
+`kafka` binding specific options.
+
+#### kafka.topics
+
+> `array` of `object`
+
+Topic configuration.
+
+<!-- @include: ../.partials/options-kafka-topics.md -->
+
+#### topics[].defaultOffset
+
+> `enum` [ `live`, `historical` ]
+
+Fetch offset to use for new consumers
+
+#### topics[].transforms
+
+> `array`
+
+<!-- todo: Dev input -->
```

Comment:
@jfallows I am not sure how to document the transforms options:

```
          transforms:
            - extract-key: ${message.key.id}
            - extract-headers:
                correlation-id: ${message.value.correlationId}
```

```
#### transforms[].extract-key

> `string`

#### transforms[].extract-headers

> `object` as map of named: `string` properties | Pattern: `^\\$\\{message\\.value\\.([A-Za-z_][A-Za-z0-9_]*)\\}$`
```

The `extract-header` property in the schema has a specified pattern but the `extract-key` doesn't. What do we need to convey to the users about this syntax and will the `${message}` syntax be used in other areas?

Reply:
@vordimous these should both have the same pattern, and they should both be able to reference either the message key or the message value.

```
> `object` as map of named: `string` properties | Pattern: `^\\$\\{message\\.(key|value)\\.([A-Za-z_][A-Za-z0-9_]*)\\}$`
```

Reply:
Added a pr to zilla to include this pattern on both properties: https://github.com/aklivity/zilla/pull/1251